sound/soc/sof/debug.c

```diff
@@ -17,6 +17,162 @@
+					 &reply, sizeof(reply));
+		end = ktime_get();
+
+		if (ret < 0)
+			break;
+
+		/* compute min and max response times */
+		ipc_response_time = ktime_to_ns(ktime_sub(end, start));
+		min_response_time = min(min_response_time, ipc_response_time);
+		max_response_time = max(max_response_time, ipc_response_time);
+
+		/* calculate the moving average */
+		avg_response_time +=
+			(ipc_response_time - avg_response_time) / (i + 1);
+		i++;
+
+		/* test complete? */
+		if (flood_duration_test) {
+			if (ktime_to_ns(end) >= test_end)
+				break;
+		} else {
+			if (i == ipc_count)
+				break;
+		}
+	}
+
+	if (ret < 0)
+		dev_err(sdev->dev,
+			"error: ipc flood test failed at %d iterations\n", i);
+
+	if (flood_duration_test)
+		dev_dbg(sdev->dev, "IPC Flood test duration: %lldms\n",
+			(long long)ipc_duration_ms);
+
+	dev_dbg(sdev->dev,
+		"IPC Flood test count: %d, Avg response time: %lldns\n",
+		i, (long long)avg_response_time);
+	dev_dbg(sdev->dev, "IPC Flood test Max response time: %lldns\n",
+		(long long)max_response_time);
+	dev_dbg(sdev->dev, "IPC Flood test Min response time: %lldns\n",
+		(long long)min_response_time);
+
+	return ret;
+}
+#endif
+
+static ssize_t sof_dfsentry_write(struct file *file, const char __user *buffer,
+				  size_t count, loff_t *ppos)
+{
+	struct snd_sof_dfsentry *dfse = file->private_data;
+	size_t size;
+	char *string;
+	int ret;
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_ENABLE_IPC_FLOOD_TEST)
+	struct snd_sof_dev *sdev = dfse->sdev;
+	unsigned long ipc_duration_ms = 0;
+	bool flood_duration_test = false;
+	unsigned long ipc_count = 0;
+	int err;
+#endif
+	string = kzalloc(count, GFP_KERNEL);
+	if (!string)
+		return -ENOMEM;
+
+	size = simple_write_to_buffer(string, count, ppos, buffer, count);
+	ret = size;
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_ENABLE_IPC_FLOOD_TEST)
+	if (!strcmp(dfse->dfsentry->d_name.name, "ipc_flood_count"))
+		goto test;
+
+	if (!strcmp(dfse->dfsentry->d_name.name, "ipc_flood_duration_ms")) {
+		flood_duration_test = true;
+		goto test;
+	}
+
+	/*
+	 * write op is only supported for ipc_flood_count or
+	 * ipc_flood_duration_ms debugfs entries atm.
+	 * ipc_flood_count floods the DSP with the number of IPC's specified.
+	 * ipc_duration_ms test floods the DSP for the time specified
+	 * in the debugfs entry.
+	 */
+	return -EINVAL;
+test:
+	ret = pm_runtime_get_sync(sdev->dev);
+	if (ret < 0) {
+		dev_err_ratelimited(sdev->dev,
+				    "error: debugfs write failed to resume %d\n",
+				    ret);
+		pm_runtime_put_noidle(sdev->dev);
+		return ret;
+	}
+
+	/* test completion criterion */
+	if (flood_duration_test)
+		ret = kstrtoul(string, 0, &ipc_duration_ms);
+	else
+		ret = kstrtoul(string, 0, &ipc_count);
+
+	if (ret < 0)
+		return ret;
+
+	/* limit max duration/ipc count for flood test */
+	if (flood_duration_test)
+		ipc_duration_ms = min(ipc_duration_ms,
+				      MAX_IPC_FLOOD_DURATION_MS);
+	else
+		ipc_count = min(ipc_duration_ms, MAX_IPC_FLOOD_COUNT);
```

Comment:
@ranj063 
Ha, I got this bug
```suggestion
		ipc_count = min(ipc_count, MAX_IPC_FLOOD_COUNT);
```

Reply:
@xiulipan thanks for spotting. Fixed now